<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>803</width>
    <height>677</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>801</width>
      <height>641</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Maze</string>
     </attribute>
     <widget class="MazeView" name="image">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>20</y>
        <width>500</width>
        <height>500</height>
       </rect>
      </property>
      <property name="baseSize">
       <size>
        <width>500</width>
        <height>500</height>
       </size>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="solve">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>540</y>
        <width>83</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>Solve maze</string>
      </property>
     </widget>
     <widget class="QPushButton" name="generateButton">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>540</y>
        <width>171</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate new maze</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="size">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>540</y>
        <width>48</width>
        <height>29</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QPushButton" name="loadMaze">
      <property name="geometry">
       <rect>
        <x>650</x>
        <y>30</y>
        <width>83</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>load maze</string>
      </property>
     </widget>
     <widget class="QPushButton" name="saveMaze">
      <property name="geometry">
       <rect>
        <x>650</x>
        <y>70</y>
        <width>83</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>save maze</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Cavity</string>
     </attribute>
     <widget class="CavityView" name="cavity">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>500</width>
        <height>500</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="nextStep">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>10</y>
        <width>83</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>next step</string>
      </property>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>530</y>
        <width>541</width>
        <height>51</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="generateManually">
         <property name="text">
          <string>Generate new cavity</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Cavity size:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="cavitySize">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Live limit:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="liveLimit">
         <property name="maximum">
          <number>7</number>
         </property>
         <property name="value">
          <number>4</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Death limit:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="deathLimit">
         <property name="maximum">
          <number>7</number>
         </property>
         <property name="value">
          <number>4</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>50</y>
        <width>261</width>
        <height>71</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>interval in milliseconds:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="timeInterval">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>2000</number>
         </property>
         <property name="singleStep">
          <number>100</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_3">
         <property name="text">
          <string>next step with interval</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>803</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MazeView</class>
   <extends>QLabel</extends>
   <header>mazeview.h</header>
  </customwidget>
  <customwidget>
   <class>CavityView</class>
   <extends>QLabel</extends>
   <header>cavityview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
